modules/system/config-wfrp4e.js

```diff
@@ -2085,7 +2089,21 @@ WFRP4E.PrepareSystemItems = function() {
             name: "WFRP4E.ConditionName.Engaged",
             flags: {
                 wfrp4e: {
-                    "value": null
+                    applicationData : {},
+                    scriptData: [
+                        {
+                            trigger: "dialog",
+                            label : "@effect.name",
+                            script : `args.abort = true
+                            ui.notifications.error(game.i18n.localize("EFFECT.ShooterEngagedError"))`,
+                            options : {
+                                dialog : {
+                                    hideScript : "return !args.weapon || args.weapon.attackType == `melee` || args.weapon.properties.qualities.pistol",
+                                    activateScript : "return args.weapon.attackType == `ranged` && !args.weapon.properties.qualities.pistol"
```

Comment:
Looks good, only thing is that I'm trying to use `weapon.isMelee/isRanged` instead of checking `attackType` directly